### DISCUSSION TITLE
Is there a way to integrate `prosemirror-math`?

### DISCUSSION THREAD

**Discussion by eykrehbein** [outline/rich-markdown-editor#424] [Q&A] [answered]
Hey,

I recently stumbled over the [prosemirror-math](https://github.com/benrbray/prosemirror-math) library, which enables users to write Latex math inside of Prosemirror. Is there a way to integrate such plugins? I tried several things, e.g. trying to insert it as `extension` prop, since the Readme says this prop "Allows additional Prosemirror plugins to be passed to the underlying Prosemirror instance.". Hasn't worked.

Am I missing something?

Best, Eyk

**Comment by tommoor** [comment, marked as answer]
Because everything is serialized to markdown you need to include `toMarkdown` and `parseMarkdown` methods, there probably needs to be a custom markdown-it plugin too – [CodeFence](https://github.com/outline/rich-markdown-editor/blob/main/src/nodes/CodeFence.ts) is a good starting point for an extension as the functionality is very similar.

I would accept a PR for integration of this module into the repo by default, math support is often asked for.

**Comment by dominiczy** [comment]
Have you made any progress on this @eykrehbein ? Maybe it can be used together with https://www.npmjs.com/package/markdown-it-texmath?
